common/lib/types/stats.js

```diff
@@ -8,6 +8,19 @@ var Stats = (function() {
 		this.refused = (values && values.refused) || 0;
 	}
 
+	function MessageCategory(values) {
+		MessageCount.call(this, values);
+		this.category = undefined;
+		if (values && values.category) {
+			this.category = { };
+			for (const [key, value] of Object.entries(values.category)) {
+				if (value) {
+					this.category[key] = new MessageCount(value);
+				}
+			}
+		}
+	}
+
```

Comment:
ably-js is ES3, I'm afraid. No const, no for-of, no `Object.entries`, no array destructuring. (Tip: if you run `grunt` before making a PR, one of the tasks is the minfier, which enforces that the input is ES3 and will warn you about some things -- but not everything (only too-new syntax, not new methods))